### PR TITLE
fix undefined hook parameter

### LINKED ISSUE #891: scenario result not passed to before/after hooks

### PR DESCRIPTION
**Pull request by charlierudolph**
resolves #891

### CODE REVIEW

**Comment by likerRr** [comment]
Can we also get access to scenario name? There can be several scenarios in one feature file and in addition to `uri+line` (where error appeared) it would be helpful to have a `name` e.g. in order to save a screenshot with the name of scenario.

**Comment by charlierudolph** [comment]
url + line uniquely identifies a scenario (or an example of a scenario outline) where the name are is not guaranteed to be unique.

**Comment by charlierudolph** [comment]
I also like matching the event protocol data. Also with some of the ideas for how to view results in the future, I think we will stop saving screenshots to disk and start attaching them and thus let the formatter embed them in an html report. I believe some html reporters already do this.

File: features/hook_parameter.feature [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    Given a file named "features/support/hooks.js" with:
+      """
+      import {defineSupportCode, Status} from 'cucumber'
+
+      defineSupportCode(({After}) => {
+        After(function(testCase) {
+          let message = testCase.sourceLocation.uri + ":" + testCase.sourceLocation.line + " "
+          if (testCase.result.status === Status.FAILED) {
+            message += "failed"
+          } else {
```
Comment (kriswill) [inline features/hook_parameter.feature]: why do you need an else?  seems like the test case only fails.

Comment (charlierudolph) [reply features/hook_parameter.feature]: Using the same After hook as the previous scenario. I'll join the two scenarios into one

**Comment by kriswill** [comment]
In reference to the scenario name, we also use it to register all of the step durations with our analytics system to track performance changes over time.  We pretty much harvest all of the metadata about the scenario and record as dimensions, then use them to filter in our charting dashboards.  I realize that maybe this type of thing might be better implemented as a reporter, but having the full fidelity of the results available in the hooks has been helpful.

**Comment by charlierudolph** [comment]
@kriswill yes please use one of the reporters for that. Ideally the support code is only used for running tests and then use reporters when you need more parts of data. The event protocol formatter gives pretty much everything

**Pull request merged**